client/mysqldump.c

```diff
@@ -6097,12 +6099,22 @@ static int do_stop_slave_sql(MYSQL *mysql_con)
 {
   MYSQL_RES *slave;
   MYSQL_ROW row;
+  // do_stop_slave_sql() should only be called once
+  DBUG_ASSERT(!stopped_replicas);
```

Comment:
This config has a problem similar to what [MDEV-17856](https://jira.mariadb.org/browse/MDEV-17856) had:
There’re two different (sets of) entry points to `do_start_slave_sql()` that passing `stopped_replicas` to it as an argument is impractical if not impossible.
A global variable is the most it could have.